rechunker/algorithm.py

```diff
@@ -125,40 +168,105 @@ def rechunking_plan(
             f"Target chunk memory ({target_chunk_mem}) exceeds max_mem ({max_mem})"
         )
 
+    if max_mem < min_mem:  # basic sanity test
+        raise ValueError(
+            "max_mem ({max_mem}) cannot be smaller than min_mem ({min_mem})"
+        )
```

Comment:
We might consider making this more strict, e.g., requiring `max_mem < 2 * min_mem`. If `min_mem` is only slightly smaller than `max_mem`, the algorithm can fail (with the `AssertionError`) due to exceeding the maximum number of allowed stages. For example, I cannot exceed `min_mem=350MB` if `max_mem=500MB` in my ERA5 exmaple.